### PR TITLE
Conditional buttons & display missions on profile view

### PR DESCRIPTION
**Pull request by DavidEspinoG**
# What is new?
In order to simplify the review process, I worked on 4 different tasks in this PR
- Add conditional "Join button"
- Add conditional "active member" tag
- Render the reserved missions into "My profile" page
- Created a component for displaying the reserved missions ( you could use it for displaying reserved rockets )

### CODE REVIEW

File: src/components/Missions/MissionsTableRow.jsx [hunk 1]
```diff
@@ -1,43 +1,51 @@
 import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+// import { useEffect } from 'react';
```
Comment (Alejandro-Bernal-M) [inline src/components/Missions/MissionsTableRow.jsx:3]: Please remove the unused code so the project will look more professional

File: src/components/Missions/MissionsTableRow.jsx [hunk 2]
```diff
@@ -1,43 +1,51 @@
...
+        <p>{description}</p>
+      </div>
+      <div className="missions-table-cell mission-button-container">
         <button
           type="button"
-          className="not-member-button"
+          className={reserved ? 'isMember-button' : 'not-member-button'}
         >
-          not a member
+          {reserved ? 'Active member' : 'not a member'}
```
Comment (Alejandro-Bernal-M) [inline src/components/Missions/MissionsTableRow.jsx:25]: Very good use of the conditional in order or display the content of the button

File: src/components/Profile/ProfileList.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      </div>
+      )}
+      {elements.map((element) => (
+        <div key={element.mission_id} className="profile-list-element">
+          <p>{element[propertyName]}</p>
+        </div>
+      ))}
+    </div>
+  </div>
+);
```
Comment (Alejandro-Bernal-M) [inline src/components/Profile/ProfileList.jsx:24]: I really like the way you conditionally displayed the content 👍

File: src/redux/missions/missionsSlice.js [hunk 4]
```diff
@@ -19,6 +19,16 @@ const initialState = {
 const missionsSlice = createSlice({
   name: 'missions',
   initialState,
+  reducers: {
+    joinMission: (state, action) => {
+      const newState = state.missions.forEach((element) => {
+        if (element.mission_id === action.payload) {
+          element.reserved = !element.reserved;
```
Comment (Alejandro-Bernal-M) [inline src/redux/missions/missionsSlice.js:26]: Great logic!

**Review by Alejandro-Bernal-M — APPROVED** [review]
# Hi @DavidEspinoG 🤝  great job!
 ### Status: 🟢 🟢 🟢 ¡Approved! 🟢 🟢 🟢 


### Optional changes:
 Please check the comments below, there is just one change, the rest are points that I specially liked about your job.

#


There is nothing more to say... it's time to merge it to dev branch 🥇

File: src/redux/missions/missionsSlice.js (hunk 4 above)
Comment (DavidEspinoG) [reply src/redux/missions/missionsSlice.js:26]: Thanks a lot for you comments, @Alejandro-Bernal-M, I think you're going to be an excellent code reviewer 👏🏼

**Pull request merged**